### DISCUSSION TITLE
App proxy showing 404 page

### DISCUSSION THREAD

**Discussion by Aswin2667** [kinngh/shopify-node-mongodb-next-app#57] [Q&A] [answered]
Hey, i tried setting up app proxy as you said in set up but i got this error 404 iam trying to POST data to my db but i got 404 error 

my App config :  
![Screenshot from 2024-04-18 13-12-05](https://github.com/kinngh/shopify-nextjs-prisma-app/assets/99728982/8cd1f694-e560-493e-8542-8edb53526721)

my Extension :
![Screenshot from 2024-04-18 13-33-37](https://github.com/kinngh/shopify-nextjs-prisma-app/assets/99728982/9efe032a-ebf5-4ab6-a779-77b06938766f)

when i hit url : https://pasta223.myshopify.com/tools/aswin-app-223   it won't hit the ngrok url instead it send req to below url 
https://pasta223.myshopify.com/api/proxy_route?shop=pasta223.myshopify.com&logged_in_customer_id=&path_prefix=%2Ftools%2Faswin-app-223&timestamp=1713424594&signature=7e57f9e94e2ae7a9a1e8961694e2458f68584051f4eb919ec93d1f8858cfeeb9

Could you kindly assist me in determining if there is anything I may have overlooked? Your help would be greatly appreciated

**Comment by kinngh** [comment, marked as answer]
Can you hit `https://pasta223.myshopify.com/tools/aswin-app-223/json` once and confirm that in your repo `/pages/api/proxy_route/json` exits? You'd be getting a 404 because there's no `/pages/api/proxy_route/index.js` to serve that request

**Reply by Aswin2667** [reply]
now it's working when i hit `https://pasta223.myshopify.com/tools/aswin-app-223/json` this URL, but 

![Screenshot from 2024-04-19 07-19-15](https://github.com/kinngh/shopify-nextjs-prisma-app/assets/99728982/1ee24ee7-6806-49fe-94ee-31efadcd9d7d)
`https://pasta223.myshopify.com/tools/aswin-app-223` this url not working but i have index.js handler in proxy_route i dont know why this happening newbie here can you tell me why?.

**Reply by kinngh** [reply]
If it's working when you hit `/json`, that means it's working as expected. Just add in your routes and you're good to go. I'm not sure why `proxy_route/index.js` wouldn't work by default, which is odd since the other route is working. I would just add other routes and work on that instead.

**Reply by Aswin2667** [reply]
Ohh ok . Thanks for your support ☺️🙏 and fast reply
